### DISCUSSION TITLE
How to upload file using axios post request?

### DISCUSSION THREAD

**Discussion by mrmarufpro** [remix-run/remix#4713] [Q&A]
I have a input `type="file"` with the `name="files"`. Now i want to get the file into the `ActionFunction` and send it through the axios post request. 
By doing so i'm getting the error `AxiosError: Data after transformation must be a string, an ArrayBuffer, a Buffer, or a Stream`. 
Please check the attached images for better understanding.

I really appreciate any help you can provide.

![image](https://user-images.githubusercontent.com/39343312/204775412-c629173e-5119-4578-a16f-955d193782c5.png)
![image](https://user-images.githubusercontent.com/39343312/204775430-4a60ad29-d61f-478e-9e80-dabcc8ab4a58.png)
![image](https://user-images.githubusercontent.com/39343312/204775455-2fb2faa3-61ca-4f48-88e9-0d694a062ffd.png)
![image](https://user-images.githubusercontent.com/39343312/204775471-2cddd8e6-ba1d-400f-b6f7-886329ef93b5.png)

**Comment by kiliman** [comment]
You can't directly access upload data from `formData`. You need to use the `unstable_parseMultipartFormData` API.

Once you do that, you'll have access to your `files` data to send upstream.

https://remix.run/docs/en/v1/api/remix#unstable_parsemultipartformdata

**Reply by mrmarufpro** [reply]
Looks like a solution for me.

By doing so I got the `data` which is a `AsyncIterable<Uint8Array>` type value.  Now I need help with how I can upload this `data` with `axios.post` request. 

I have tried to add the data to axios body like `new FormData().append("files",data)` which resulting `AsyncIterable<Uint8Array>' is not assignable to parameter of type 'string | Blob'`

Can you help me with another blog or something please.

![image](https://user-images.githubusercontent.com/39343312/204928075-1ac9e9d3-c696-4b1e-8d68-bd1ac72e91fc.png)

**Comment by a-atalla** [comment]
Trying the same with fetch and i get `400 Bad Request and Multipart: Boundary not found in C-T.`

also this uploadHandler doesn't work
```typescript
 const uploadHandler = async ({ name, contentType, data, filename }) => {
    // Get the file as a buffer
    const chunks = [];
    for await (const chunk of data) chunks.push(chunk);
    const buffer = Buffer.concat(chunks);
    const blob = new Blob([buffer]);
    return blob;
  };
  const formData = await unstable_parseMultipartFormData(
    request,
    uploadHandler
  );
```
